model/validation.go

```diff
@@ -204,16 +202,453 @@ func (t Tags) Validate() error {
+	}
+
+	var retVal error
+
+	if len(parts) == 2 {
+		if err := validateCIDRBlock(parts[1]); err != nil {
+			retVal = multierror.Append(retVal, err)
+		}
+	}
+
+	if err := validateIPv4Address(parts[0]); err != nil {
+		retVal = multierror.Append(retVal, err)
+	}
+
+	return retVal
+}
+
+// validateCIDRBlock validates that a string in "CIDR notation" or "Dot-decimal notation"
+func validateCIDRBlock(cidr string) error {
+	if bits, err := strconv.Atoi(cidr); err != nil || bits <= 0 || bits > 32 {
+		return fmt.Errorf("/%v is not a valid CIDR block", cidr)
+	}
+
+	return nil
+}
+
+// validateIPv4Address validates that a string in "CIDR notation" or "Dot-decimal notation"
+func validateIPv4Address(addr string) error {
+	octets := strings.Split(addr, ".")
+	if len(octets) != 4 {
+		return fmt.Errorf("%q is not a valid IP address", addr)
+	}
+
+	for _, octet := range octets {
+		if n, err := strconv.Atoi(octet); err != nil || n < 0 || n > 255 {
+			return fmt.Errorf("%q is not a valid IP address", addr)
+		}
+	}
+
+	return nil
+}
+
+func validateDelay(delay *proxyconfig.HTTPFaultInjection_Delay) error {
+	var retVal error
+
+	retVal = validateFloatPercent(retVal, delay.Percent, "delay")
+
+	if delay.GetFixedDelaySeconds() < 0 {
+		retVal = multierror.Append(retVal, fmt.Errorf("delay fixed_seconds invalid"))
+	}
+
+	if delay.GetExponentialDelaySeconds() != 0 {
+		if delay.GetExponentialDelaySeconds() < 0 {
+			retVal = multierror.Append(retVal, fmt.Errorf("delay exponential_seconds invalid"))
+		}
+		retVal = multierror.Append(retVal, fmt.Errorf("exponential_seconds not implemented"))
+	}
+
+	return retVal
+}
+
+func validateAbortHTTPStatus(httpStatus *proxyconfig.HTTPFaultInjection_Abort_HttpStatus) error {
+	var retVal error
+
+	if httpStatus.HttpStatus < 0 || httpStatus.HttpStatus > 600 {
+		retVal = multierror.Append(retVal, fmt.Errorf("invalid abort http status %v", httpStatus.HttpStatus))
+	}
+
+	return retVal
+}
+
+func validateAbort(abort *proxyconfig.HTTPFaultInjection_Abort) error {
+	var retVal error
+
+	retVal = validateFloatPercent(retVal, abort.Percent, "abort")
+
+	switch abort.ErrorType.(type) {
+	case *proxyconfig.HTTPFaultInjection_Abort_GrpcStatus:
+		// No validation yet for grpc_status / http2_error / http_status
+	case *proxyconfig.HTTPFaultInjection_Abort_Http2Error:
+		// No validation yet for grpc_status / http2_error / http_status
+	case *proxyconfig.HTTPFaultInjection_Abort_HttpStatus:
+		if err := validateAbortHTTPStatus(abort.ErrorType.(*proxyconfig.HTTPFaultInjection_Abort_HttpStatus)); err != nil {
+			retVal = multierror.Append(retVal, err)
+		}
+	}
+
+	// No validation yet for override_header_name
+
+	return retVal
+}
+
+func validateTerminate(terminate *proxyconfig.L4FaultInjection_Terminate) error {
+	var retVal error
+
+	retVal = validateFloatPercent(retVal, terminate.Percent, "terminate")
+
+	return retVal
+}
+
+func validateThrottle(throttle *proxyconfig.L4FaultInjection_Throttle) error {
+	var retVal error
+
```

Comment:
Sam comment as before.  The Throttle is correct, but we don't support it, so I have put the check on the caller.